akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -163,26 +164,43 @@ abstract class ActorMaterializer extends Materializer {
   def isShutdown: Boolean
 
   /**
-   * INTERNAL API: this might become public later
+   * INTERNAL API
    */
-  private[akka] def actorOf(context: MaterializationContext, props: Props): ActorRef
+  def actorOf(context: MaterializationContext, props: Props): ActorRef
 
   /**
    * INTERNAL API
    */
-  private[akka] def system: ActorSystem
+  def system: ActorSystem
 
   /**
    * INTERNAL API
    */
   private[akka] def logger: LoggingAdapter
 
-  /** INTERNAL API */
+  /**
+   * INTERNAL API
+   */
   private[akka] def supervisor: ActorRef
 
 }
 
 /**
+ * ExtendedActorMaterializer used by subtypes which materializer using GraphInterpreterShell
+ */
+abstract class ExtendedActorMaterializer extends ActorMaterializer {
```

Comment:
shall we place this in the `impl` package for now, since it also requires `GraphInterpreterShell` that is in that package?